### PR TITLE
Client DC Checking for new_player

### PR DESCRIPTION
**Pull request by ZephyrTFA**
## About The Pull Request

Don't you love seeing this
```
[2023-01-08 19:43:47.400] runtime error: Cannot execute null.set db player flags().
 - proc name: Login (/mob/dead/new_player/Login)
 -   source file: login.dm,7
 -   usr: (src)
 -   src: xxx (/mob/dead/new_player)
 -   src.loc: null
 -   call stack:
 - xxx (/mob/dead/new_player): Login()
 - 
[2023-01-08 19:43:47.453] runtime error: Cannot execute null.set db player flags().
 - proc name: Login (/mob/dead/new_player/Login)
 -   source file: login.dm,7
 -   usr: (src)
 -   src: xxx (/mob/dead/new_player)
 -   src.loc: null
 -   call stack:
 - xxx (/mob/dead/new_player): Login()
 - 
[2023-01-08 19:43:47.573] runtime error: Tried to set a mind's current var to a qdeleted mob, what the fuck
 - proc name: set current (/datum/mind/proc/set_current)
 -   source file: _mind.dm,140
 -   usr: xxx (/mob/dead/new_player)
 -   src: /datum/mind (/datum/mind)
 -   usr.loc: null
 -   call stack:
 - /datum/mind (/datum/mind): set current(Phaseu (/mob/dead/new_player))
 - xxx (/mob/dead/new_player): Login()
 - 
[2023-01-08 19:43:47.573] runtime error: Cannot read null.holder
 - proc name: Login (/mob/dead/new_player/Login)
 -   source file: login.dm,14
 -   usr: (src)
 -   src: xxx (/mob/dead/new_player)
 -   src.loc: null
 -   call stack:
 - xxx (/mob/dead/new_player): Login()
```
I dont.
## Why It's Good For The Game

Less runtimes.

### CODE REVIEW

File: code/modules/mob/dead/new_player/login.dm [hunk 1]
```diff
@@ -3,8 +3,12 @@
 		return
 
 	if(CONFIG_GET(flag/use_exp_tracking))
-		client.set_exp_from_db()
-		client.set_db_player_flags()
+		client?.set_exp_from_db()
+		client?.set_db_player_flags()
+		if(!client)
```
Comment (Cyprex) [inline code/modules/mob/dead/new_player/login.dm:8]: Why not put this check at the start of of this bock, what's the point of calling the two functions if client is already null

Comment (ZephyrTFA) [reply code/modules/mob/dead/new_player/login.dm:8]: because these calls can lead to the client becoming null.

read the doc.

**Comment by optimumtact** [comment]
this isn't checking at all though is it, it's just blindly supressing the messages

**Comment by ZephyrTFA** [comment]
> this isn't checking at all though is it, it's just blindly supressing the messages

It's checking for the client being nulled out, see client disconnection, and aborting the Login procedure

File: code/modules/mob/dead/new_player/login.dm [hunk 2]
```diff
@@ -3,8 +3,12 @@
 		return
 
 	if(CONFIG_GET(flag/use_exp_tracking))
-		client.set_exp_from_db()
-		client.set_db_player_flags()
+		client?.set_exp_from_db()
+		client?.set_db_player_flags()
```
Comment (MrStonedOne) [inline code/modules/mob/dead/new_player/login.dm:7]: both of these should have been moved to client/New() instead. on-connection events should not be happening in Login()

Comment (ZephyrTFA) [reply code/modules/mob/dead/new_player/login.dm:7]: I can move them aye

**Pull request merged**